ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/SupportsProtocol.hs

```diff
@@ -99,13 +99,19 @@ instance
   LedgerSupportsProtocol (ShelleyBlock (Praos crypto) era)
   where
   protocolLedgerView _cfg st =
-    let SL.NewEpochState {nesPd, nesEs} = tickedShelleyLedgerState st
+    let nes = tickedShelleyLedgerState st
+
+        SL.NewEpochState {nesPd} = nes
+
+        pparam :: forall a. Lens.Micro.Lens' (LedgerCore.PParams era) a -> a
+        pparam lens = getPParams nes Lens.Micro.^. lens
```

Comment:
I rather use `^.` since there are way too many `.`s already :) But it's just a personal pref.